src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs

```diff
@@ -567,7 +604,18 @@ private void CloseButton_OnClick(object sender, RoutedEventArgs e)
         // Handles Close button on the Warning bar
         private void CloseWarningBarButton_OnClick(object sender, RoutedEventArgs e)
         {
-            this.Close();
+            var senderContext = (sender as Button).DataContext;
+
+            if (senderContext.Equals(this))
+            {
+                Close();
+            }
+            else // Close the right side extension tab if the close button is clicked on the docked editor. 
+            {
+                var dynamoView = Owner as DynamoView;
+                TabItem tabItem = dynamoView.ExtensionTabItems.OfType<TabItem>().SingleOrDefault(n => n.Uid.ToString() == NodeModel.GUID.ToString());
+                dynamoView.CloseExtensionTab(tabItem);
```

Comment:
Why do we close the whole tab as well?

Reply:
There is a close button on the warning message and that would close the tab without saving. Like how the script editor window is closed.